engines/sherlock/objects.h

```diff
@@ -0,0 +1,271 @@
+	PAL_CHANGE2	= 10, // Same as PAL_CHANGE, except that it goes up
+	SCRIPT_ZONE	= 11, // If this is clicked in, it is activated
+	BLANK_ZONE	= 12, // This masks out other objects when entered
+	NOWALK_ZONE = 13  // Player cannot walk here
+};
+
+// Different levels for sprites to be at
+enum {
+	BEHIND = 0, NORMAL_BEHIND = 1, NORMAL_FORWARD = 2, FORWARD = 3
+};
+
+#define MAX_HOLMES_SEQUENCE 16
+#define MAX_FRAME 30
+
+// code put into sequences to defines 1-10 type seqs
+#define SEQ_TO_CODE 67
+#define FLIP_CODE (64 + 128)
+#define SOUND_CODE (34 + 128)
+
+class Point32 {
+public:
+	int x;
+	int y;
+
+	Point32() : x(0), y(0) {}
+	Point32(int x1, int y1) : x(x1), y(y1) {}
+	Point32(const Common::Point &pt) : x(pt.x), y(pt.y) {}
+
+	bool operator==(const Point32 &p) const { return x == p.x && y == p.y; }
+	bool operator!=(const Point32 &p) const { return x != p.x || y != p.y; }
+	Point32 operator+(const Point32 &delta) const { return Point32(x + delta.x, y + delta.y); }
+	Point32 operator-(const Point32 &delta) const { return Point32(x - delta.x, y - delta.y); }
+	operator Common::Point() { return Common::Point(x, y); }
+
+	void operator+=(const Point32 &delta) { x += delta.x; y += delta.y; }
+	void operator-=(const Point32 &delta) { x -= delta.x; y -= delta.y; }
+};
+
+class Sprite {
+private:
+	static SherlockEngine *_vm;
+public:
+	Common::String _name;				// Name
+	Common::String _description;		// Description
+	Common::StringArray _examine;		// Examine in-depth description
+	Common::String _pickUp;				// Message for if you can't pick up object
+
+	const uint8 (*_sequences)[MAX_HOLMES_SEQUENCE][MAX_FRAME];  // Holds animation sequences
+	ImageFile *_images;					// Sprite images
+	ImageFrame *_imageFrame;			// Pointer to shape in the images
+	int _walkCount;						// Character walk counter
+	int _allow;							// Allowed menu commands - ObjectAllow
+	int _frameNumber;					// Frame number in rame sequence to draw
+	int _sequenceNumber;				// Sequence being used
+	Point32 _position;					// Current position
+	Point32 _delta;						// Momvement delta
+	Common::Point _oldPosition;			// Old position
+	Common::Point _oldSize;				// Image's old size
+	Common::Point _goto;				// Walk destination
+	SpriteType _type;					// Type of object
+	int _pickup;
+	Common::Point _noShapeSize;			// Size of a NO_SHAPE
+	int _status;						// Status: open/closed, moved/not moved
+	int8 _misc;							// Miscellaneous use
+	int _numFrames;						// How many frames the object has
+public:
+	Sprite() { clear(); }
+	static void setVm(SherlockEngine *vm) { _vm = vm; }
+
+	void clear();
+
+	void setImageFrame();
+
+	void adjustSprite();
+
+	void checkSprite();
+
+	int frameWidth() const { return _imageFrame ? _imageFrame->_frame.w : 0; }
+	int frameHeight() const { return _imageFrame ? _imageFrame->_frame.h : 0; }
+};
+
+enum { REVERSE_DIRECTION = 0x80 };
+
+struct ActionType {
+	int _cAnimNum;
+	int _cAnimSpeed;				// if high bit set, play in reverse
```

Comment:
You're right. It was that way originally, but I standardized on using a negative number for reverse, since it was being done that way in other places.